inspirehep/modules/multieditor/api.py

```diff
@@ -0,0 +1,147 @@
+        return jsonify({'message': 'Please use the search before you apply actions'}), 400
+
+    uuids, records = queries.get_paginated_records(page_number=page_number,
+                                                   page_size=page_size,
+                                                   uuids=multieditor_session['uuids'])
+
+    old_records = copy.deepcopy(records)
+    try:
+        actions = get_actions(user_actions, multieditor_session['schema'])
+    except InvalidActions as err:
+        return jsonify({'message': err.message}), 400
+    for record in records:
+        for action in actions:
+            try:
+                action.process(record, multieditor_session['schema'])
+            except (SchemaError, InvalidValue) as err:
+                return jsonify({'message': err.message}), 400
+    json_patches, errors = compare_records(old_records, records, multieditor_session['schema'])
+
+    return jsonify({'json_records': old_records, 'errors': errors, 'json_patches': json_patches, 'uuids': uuids})
+
+
+@blueprint.route("/paginate", methods=['GET'])
+@multieditor_use_permission.require(http_exception=403)
+def paginate():
+    """Get paginated records from the session"""
+    page_number = int(request.args.get('pageNum', 1))
+    page_size = int(request.args.get('pageSize', 10))
+    multieditor_session = session.get('multieditor_session', {})
+    if not multieditor_session:
+        return jsonify({'message': 'Please refresh your page'}), 400
+    uuids, records = queries.get_paginated_records(page_number=page_number,
+                                                   page_size=page_size, uuids=multieditor_session['uuids'])
+    return jsonify({'uuids': uuids, 'json_records': records})
+
+
+@blueprint.route("/search", methods=['GET'])
+@multieditor_use_permission.require(http_exception=403)
+def search():
+    """Search for records using the query and store the result's ids"""
+    query_string = request.args.get('queryString', '')
+    page_number = int(request.args.get('pageNum', 1))
+    page_size = int(request.args.get('pageSize', 10))
+    index_name = request.args.get('index', '')
+    schema = load_schema(index_name, resolved=True)
+    total_records = queries.get_total_records(query_string, index_name)
+    if total_records > current_app.config['MULTI_MAX_RECORDS']:
+        return jsonify({'message': 'Please narrow the results using a query to be less than 10000'}), 400
+    uuids = queries.get_record_ids_from_query(query_string, index_name)
+    session['multieditor_session'] = {
+        'uuids': uuids,
+        'schema': schema,
+    }
+    records_uuids, records = queries.get_paginated_records(page_number=page_number,
+                                                           page_size=page_size,
+                                                           uuids=uuids)
+    return jsonify({'total_records': total_records,
+                    'uuids': records_uuids,
+                    'json_records': records})
```

Comment:
For the REST API we can use ``MethodView`` provided from Flask, we can separate the resources based on the logic, i.e.

```python
class MultiEditorRecordItemResource(MethodView):

  def post(self):
      """...."""

multieditor_record_item = MultiEditorRecordItemResource('multieditor_record_item')
blueprint.add_url_rule(
    '/multieditor/record',
    view_func= multieditor_record_item,
)
```